### ISSUE TITLE
fix login erorr

### ISSUE THREAD

**Issue by manhkhac** [Schmavery/facebook-chat-api#890]
fix login erorr

**Comment by coreyjansen** [comment]
I am getting the error 
```
info login Logging in...
ERR! login Wrong username/password.
{ error: 'Wrong username/password.' }
```

**Comment by manhkhac** [comment]
I mean help me fix my login error

**Comment by ram-web-developer** [comment]
info login Logging in...
ERR! login Wrong username/password.
{ error: 'Wrong username/password.' }

  please fix this

**Comment by lv185039** [comment]
This error usually means just that, are you able to login to the account on a browser with the same details? Do you have them properly defined in the project using this plugin as such: 

`login({email: "FB_EMAIL", password: "FB_PASSWORD"}, (err, api) => {()}`

**Comment by RamPeregrine** [comment]
> This error usually means just that, are you able to login to the account on a browser with the same details? Do you have them properly defined in the project using this plugin as such:
> 
> `login({email: "FB_EMAIL", password: "FB_PASSWORD"}, (err, api) => {()}`


yes im able to login  on a browser with the same details, but in code this not work and showing  this 
error { error: 'Wrong username/password.' }

**Comment by LinkTree3** [comment]
Same issue for me right now.
Able to login with the browser and have set the login function input object exactly like this.
`login({email: "FB_EMAIL", password: "FB_PASSWORD"}, (err, api) => {()}`

Has anyone had any success with this yet?

**Comment by nktnet1** [comment]
This video contains a workaround:
https://www.youtube.com/watch?v=y9_yd5a3scM
(turn on subtitle)

Basically:
1. Extract cookies from facebook.com e.g. by using 
    - https://chrome.google.com/webstore/detail/j2team-cookies/okpidcojinmlaakglciglbpcpajaibco/related
    
2. Copy the value of the key `cookies` (should be an array of objects) and put the array in a file, e.g. `appstate.json`

3. In `appstate.json`, replace all instances of `name` with `key`.

4. Login:
    ```
    const fs = require('fs');
    const login = require("facebook-chat-api");
    const credential = { 
        appState: JSON.parse(fs.readFileSync('appstate.json', 'utf-8'))
    }
    login(
      credential,
      (error, api) => {
          // ...
      }
    );
    ```

**Comment by lukman-fun** [comment]
> This video contains a workaround: https://www.youtube.com/watch?v=y9_yd5a3scM (turn on subtitle)
> 
> Basically:
> 
> 1. Extract cookies from facebook.com e.g. by using
>    
>    * https://chrome.google.com/webstore/detail/j2team-cookies/okpidcojinmlaakglciglbpcpajaibco/related
> 2. Copy the value of the key `cookies` (should be an array of objects) and put the array in a file, e.g. `appstate.json`
> 3. In `appstate.json`, replace all instances of `name` with `key`.
> 4. Login:
>    ```
>    const fs = require('fs');
>    const login = require("facebook-chat-api");
>    const credential = { 
>        appState: JSON.parse(fs.readFileSync('appstate.json', 'utf-8'))
>    }
>    login(
>      credential,
>      (error, api) => {
>          // ...
>      }
>    );
>    ```

appState.map not function

**Comment by sebastian-paul** [comment]
> > This video contains a workaround: https://www.youtube.com/watch?v=y9_yd5a3scM (turn on subtitle)
> > Basically:
> > 
> > 1. Extract cookies from facebook.com e.g. by using
> >    
> >    * https://chrome.google.com/webstore/detail/j2team-cookies/okpidcojinmlaakglciglbpcpajaibco/related
> > 2. Copy the value of the key `cookies` (should be an array of objects) and put the array in a file, e.g. `appstate.json`
> > 3. In `appstate.json`, replace all instances of `name` with `key`.
> > 4. Login:
> >    ```
> >    const fs = require('fs');
> >    const login = require("facebook-chat-api");
> >    const credential = { 
> >        appState: JSON.parse(fs.readFileSync('appstate.json', 'utf-8'))
> >    }
> >    login(
> >      credential,
> >      (error, api) => {
> >          // ...
> >      }
> >    );
> >    ```
> 
> appState.map not function

update appState with cookies value like below 

const credential = {
    appState: JSON.parse(fs.readFileSync('appstate.json', 'utf-8')).cookies
}
